pkg/common/hubconfig/hubconfig.go

```diff
@@ -65,22 +65,21 @@ func PrepareHubConfig(tlsClientInsecure bool) (*rest.Config, error) {
 			},
 		}
 	} else {
+		var caData []byte
 		hubCA, err := env.Lookup(hubCAEnvKey)
-		if err != nil {
-			klog.ErrorS(err, "Hub certificate authority cannot be empty")
-			return nil, err
-		}
-		decodedClusterCaCertificate, err := base64.StdEncoding.DecodeString(hubCA)
-		if err != nil {
-			klog.ErrorS(err, "Cannot decode hub cluster certificate authority data")
-			return nil, err
+		if err == nil {
```

Comment:
why is that the err != nil not checked?

Reply:
```suggestion
                // use the agent OS's root CA when the hubCA env is not set
		if err == nil { // check if hubCA env is not set
```

Please add the comment here to improve the readability as in normal case we check the err != nil and need some signal here to indicate why we check err == nil here.